File: Gixor-Backend/Gixor-Api/Gixor-Api-GitHub/src/main/java/com/cpy3f2/Gixor/Domain/DTO/SubscriptionDTO.java
package com.cpy3f2.Gixor.Domain.DTO;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.time.LocalDateTime;

/**
 *  SubscriptionDTO
 * @author simon
 * @since 2024/11/4 */
@Data
public class SubscriptionDTO {
    private Boolean subscribed;
    private Boolean ignored;
    private String reason;
    
    @JsonProperty("created_at")
    private LocalDateTime createdAt;
    
    private String url;
    
    @JsonProperty("repository_url")
    private String repositoryUrl;
} 

File: Gixor-Mobile/app/src/main/java/com/cpy3f2/gixor_mobile/model/entity/SubscriptionDTO.kt
package com.cpy3f2.gixor_mobile.model.entity

import java.time.LocalDateTime

data class SubscriptionDTO(
    val subscribed: Boolean? = null,
    val ignored: Boolean? = null,
    val reason: String? = null,

    val createdAt: LocalDateTime? = null,
    
    val url: String? = null,

    val repositoryUrl: String? = null
)